Feature: add-email custom headers

jeffa5 opened the pull request:
Fixes: https://github.com/scheibler/khard/issues/257

This adds new arguments to the `add-email` sub command, `-H`, `--header` and `--all`, which allow extracting the email addresses from the given header, or all of the headers if requested.

This also adds a loop over the found email addresses so that the user is given the opportunity to add each to a contact.

A later extension of this might allow `-H` and `--header` to accept a list of headers to use, but this isn't implemented here.

lucc commented on khard/khard.py:
```diff
@@ -543,37 +544,27 @@ def new_subcommand(selected_address_books: AddressBookCollection,
         create_new_contact(selected_address_book)
 
+def add_email_to_contact(name: str, email_address: str, abooks: AddressBookCollection) -> None:
+    """Add a new email address to the given contact, creating the contact if necessary.
```
PEP-8 line length please.

lucc commented on test/test_khard.py:
```diff
@@ -44,3 +46,39 @@ def test_no_search_terms_result_in_any_queries(self):
                              source_search_terms=query.AnyQuery(),
                              target_contact=query.AnyQuery())
         self.assertEqual(expected, prepared["foo"])
+
```
PEP-8 line length please.

jeffa5 replied:
Yep, is there a standard auto-formatter you'd recommend for this project?

lucc replied:
No, I have a language server for python in my editor and have a marker for overlong lines.

jeffa5 commented on khard/cli.py:
```diff
@@ -278,6 +278,12 @@ def create_parsers() -> Tuple[argparse.ArgumentParser,
+        )
+    add_email_parser.add_argument(
+        "--all", dest="field", action="store_const", const="all", help="Extract contacts from all headers"
+        )
```
Sure, I'll run a formatter over it.

Perhaps we could just have the `-H` option for now with the comment in the docs. I'd find it easier adding a new argument later rather than removing one.

Yes, I'm just having a play around with the cli parsing now. I've just checked the field fetching code and it converts both to lower before comparison so we should be safe here.

jeffa5 commented:
I've removed the `--all` and implemented the multiple fields for `-H`. If one of the fields is `all` then all headers are searched. This also lower cases all fields given to aid in error messages.
